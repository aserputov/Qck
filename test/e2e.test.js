/* eslint-disable no-undef */
const { run } = require("./run");

describe("end-to-end integration", () => {
  test("prints error and help message when no arguments given", async () => {
    const { stderr, stdout, exitCode } = await run();
    const errorMsg =
      "error: Option isn't correct. Please provide correct input";
    expect(exitCode === 0).toBeFalsy();
    expect(stderr.includes(errorMsg)).toBeTruthy();
    expect(stdout).toEqual("");
  });

  test("prints help message when --help given", async () => {
    const { stderr, stdout, exitCode } = await run("--help");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints version message when --version given", async () => {
    const { stderr, stdout, exitCode } = await run("-v");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("single valid text file input should generate an html", async () => {
    const { stderr, stdout, exitCode } = await run("--input", "same.txt");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("single valid text file input should generate an html", async () => {
    const { stderr, stdout, exitCode } = await run(
      "--input",
      "same.txt",
      "--stylesheet",
      "link"
    );
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("single valid json configuration file should generate an html", async () => {
    const { stderr, stdout, exitCode } = await run("--config", "src/conf.json");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });
});
