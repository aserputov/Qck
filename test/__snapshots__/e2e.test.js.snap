// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints help message when --help given 1`] = `
"Usage: -h <file>

Options:
      --help    Show help                                              [boolean]
  -i, --input   File name                                               [string]
  -c, --config  Configuration file                                      [string]
  -v            version                                                [boolean]"
`;

exports[`end-to-end integration prints version message when --version given 1`] = `"qck-ssg v0.1.2"`;

exports[`end-to-end integration single valid json configuration file should generate an html 1`] = `"-- Link to Page Created In Index File"`;

exports[`end-to-end integration single valid text file input should generate an html 1`] = `"-- Link to Page Created In Index File"`;

exports[`end-to-end integration single valid text file input should generate an html 2`] = `"-- Link to Page Created In Index File"`;
